// better-auth/dist/db/index.mjs
export { d as convertFromDB, b as convertToDB, c as createInternalAdapter, a as getAdapter, e as getMigrations, f as getSchema, g as getWithHooks, m as matchType } from '../shared/better-auth.Bb80Qhal.mjs';
export { g as getAuthTables } from '../shared/better-auth.Cjw1GDsh.mjs';
export { t as toZodSchema } from '../shared/better-auth.BxexnJiR.mjs';
export { a as accountSchema, c as coreSchema, g as getAllFields, m as mergeSchema, j as parseAccountInput, d as parseAccountOutput, i as parseAdditionalUserInput, f as parseInputData, p as parseOutputData, k as parseSessionInput, e as parseSessionOutput, h as parseUserInput, b as parseUserOutput, s as sessionSchema, u as userSchema, v as verificationSchema } from '../shared/better-auth.Dlx1jCB2.mjs';
import '../shared/better-auth.CW6D9eSx.mjs';
import '../shared/better-auth.Cwj5CPkV.mjs';
import '../shared/better-auth.CiuwFiHM.mjs';
import '../shared/better-auth.CvNZNAOW.mjs';
import '../shared/better-auth.C4g12FAA.mjs';
import '../shared/better-auth.BUPPRXfK.mjs';
import '@better-auth/utils/random';
import 'zod';
import 'better-call';
import '@better-auth/utils/hash';
import '@noble/ciphers/chacha.js';
import '@noble/ciphers/utils.js';
import '@better-auth/utils/base64';
import 'jose';
import '@noble/hashes/scrypt.js';
import '@better-auth/utils/hex';
import '@noble/hashes/utils.js';
import '../shared/better-auth.B4Qoxdgc.mjs';
import '../shared/better-auth.DdzSJf-n.mjs';
import '../shared/better-auth.DUTxy-hg.mjs';
import 'kysely';
import '../shared/better-auth.X9kq8Nfb.mjs';
import '../shared/better-auth.AYS43XM1.mjs';

const createFieldAttribute = (type, config) => {
  return {
    type,
    ...config
  };
};

export { createFieldAttribute };
