// better-auth/dist/integrations/react-start.mjs
import '../shared/better-auth.CiuwFiHM.mjs';
import '@better-auth/utils/base64';
import '@better-auth/utils/hmac';
import '../shared/better-auth.C4g12FAA.mjs';
import '@better-auth/utils/binary';
import { p as parseSetCookieHeader } from '../shared/better-auth.pyHheoY8.mjs';
import 'better-call';
import 'zod';
import { c as createAuthMiddleware } from '../shared/better-auth.CIqzVuC5.mjs';
import '../shared/better-auth.DlNonChL.mjs';
import '../shared/better-auth.Dlx1jCB2.mjs';
import '../shared/better-auth.8GJtw9Eq.mjs';
import '../plugins/organization/access/index.mjs';
import '@better-auth/utils/random';
import '@better-auth/utils/hash';
import '@noble/ciphers/chacha.js';
import '@noble/ciphers/utils.js';
import 'jose';
import '@noble/hashes/scrypt.js';
import '@better-auth/utils/hex';
import '@noble/hashes/utils.js';
import '../shared/better-auth.B4Qoxdgc.mjs';
import 'kysely';
import '@better-auth/utils/otp';
import '../plugins/admin/access/index.mjs';
import '@better-fetch/fetch';
import '@better-auth/utils';
import '../plugins/custom-session/index.mjs';
import '@noble/hashes/sha3.js';
import '../plugins/device-authorization/index.mjs';
import '../shared/better-auth.DdzSJf-n.mjs';
import '../shared/better-auth.CW6D9eSx.mjs';
import '../shared/better-auth.CvNZNAOW.mjs';
import '../shared/better-auth.BAQSo96z.mjs';
import '../shared/better-auth.D2xndZ2p.mjs';
import '../crypto/index.mjs';
import 'jose/errors';
import '../shared/better-auth.BUPPRXfK.mjs';
import 'defu';
import '../plugins/access/index.mjs';
import '../shared/better-auth.DQI8AD7d.mjs';
import '../shared/better-auth.BpA03GIs.mjs';

const reactStartCookies = () => {
  return {
    id: "react-start-cookies",
    hooks: {
      after: [
        {
          matcher(ctx) {
            return true;
          },
          handler: createAuthMiddleware(async (ctx) => {
            const returned = ctx.context.responseHeaders;
            if ("_flag" in ctx && ctx._flag === "router") {
              return;
            }
            if (returned instanceof Headers) {
              const setCookies = returned?.get("set-cookie");
              if (!setCookies) return;
              const parsed = parseSetCookieHeader(setCookies);
              const { setCookie } = await import('../chunks/index.mjs');
              parsed.forEach((value, key) => {
                if (!key) return;
                const opts = {
                  sameSite: value.samesite,
                  secure: value.secure,
                  maxAge: value["max-age"],
                  httpOnly: value.httponly,
                  domain: value.domain,
                  path: value.path
                };
                try {
                  setCookie(key, decodeURIComponent(value.value), opts);
                } catch (e) {
                }
              });
              return;
            }
          })
        }
      ]
    }
  };
};

export { reactStartCookies };
